### PR TITLE
Continue removing non-references codes

### PR DESCRIPTION
**Pull request by brycewang-microsoft**
Removing non-references codes for the folders `provisioning` and `security` in particular. No public-facing API changes.

### CODE REVIEW

File: provisioning/service/src/Enrollments/AttestationMechanism.cs [hunk 1]
```diff
@@ -48,42 +48,6 @@ internal AttestationMechanism(Attestation attestation)
             }
         }
 
-        [JsonConstructor]
```
Comment (drwill-ms) [inline provisioning/service/src/Enrollments/AttestationMechanism.cs:51]: How do we know this one is safe to remove? It is a constructor for JSON so it wouldn't have references.

File: provisioning/transport/http/src/Generated/RuntimeRegistrationExtensions.cs [hunk 2]
```diff
@@ -12,29 +12,6 @@ namespace Microsoft.Azure.Devices.Provisioning.Client.Transport
     /// </summary>
     internal static partial class RuntimeRegistrationExtensions
     {
-        /// <summary>
```
Comment (drwill-ms) [inline provisioning/transport/http/src/Generated/RuntimeRegistrationExtensions.cs:15]: Should we avoid editing generated files?

File: provisioning/transport/http/src/SymmetricKeyCredentials.cs [hunk 3]
```diff
@@ -12,30 +12,11 @@ namespace Microsoft.Azure.Devices.Provisioning.Client.Transport
...
-        private const string Registration = "registration";
         private readonly string SymmetricKey;
-        private volatile string _sasToken;
 
         public SymmetricKeyCredentials(string symmetricKey) : base()
         {
             SymmetricKey = symmetricKey;
         }
-
-        public override Task ProcessHttpRequestAsync(HttpRequestMessage request, CancellationToken cancellationToken)
```
Comment (drwill-ms) [inline provisioning/transport/http/src/SymmetricKeyCredentials.cs:25]: I'm not sure what the bar is for removing code in this PR. Is it simply that a member doesn't have any references? I'm worried we're going too far in some cases.

File: provisioning/service/src/Enrollments/AttestationMechanism.cs (hunk 1 above)
Comment (brycewang-microsoft) [reply provisioning/service/src/Enrollments/AttestationMechanism.cs:51]: Yeah, the failed e2e tests complained about the same. I will undo the `JsonConstructor` removing.

**Review by timtay-microsoft — APPROVED** [review]

File: provisioning/transport/http/src/Generated/RuntimeRegistrationExtensions.cs (hunk 2 above)
Comment (drwill-ms) [reply provisioning/transport/http/src/Generated/RuntimeRegistrationExtensions.cs]: Yeah.

**Review by drwill-ms — APPROVED** [review]

**Pull request merged**